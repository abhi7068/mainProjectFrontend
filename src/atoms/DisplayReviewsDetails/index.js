import React from 'react';
import Rating from '@material-ui/lab/Rating';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 1800,
    textAlign: 'left',
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

export default function DisplayReviewsDetails(props) {
  const classes = useStyles();
  console.log('');
  function time() {
    var d = new Date(Date.parse(props.data.created_at)).toString();

    var index = d.lastIndexOf(':') - 5;
    return d.substring(0, index);
  }
  return (
    <Card className={classes.root}>
      <CardHeader
        style={{ color: 'black' }}
        avatar={
          <Avatar aria-label="recipe" className={classes.avatar}>
            {/* {props.data.user.profilePicture} */}
          </Avatar>
        }
        title={props.data.user.name}
        subheader={time()}
      />

      <CardContent>
        {props.data.rating !== 0 ? (
          <Rating
            name="simple-controlled"
            // style={{ textAlign: 'center' }}
            style={{ color: '#E40046' }}
            value={props.data.rating}
            readOnly
          />
        ) : (
          ''
        )}

        <Typography
          style={{ fontSize: '1.5em', textAlign: 'left' }}
          color="textSecondary"
          component="p"
        >
          {props.data.comment}
        </Typography>
      </CardContent>
    </Card>
  );
}
DisplayReviewsDetails.propTypes = {
  data: PropTypes.object,
};
