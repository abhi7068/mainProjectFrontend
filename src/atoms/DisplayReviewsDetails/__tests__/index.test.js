import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import DisplayReviewsDetails from '../index';
import React from 'react';
afterEach(cleanup);
it('renders', () => {
  const data = {
    rating: 5,
    comment: 'Nice Picture',
    user: {
      name: 'Madan Murari',
    },
  };
  const { asFragment } = render(<DisplayReviewsDetails data={data} />);
  expect(asFragment()).toMatchSnapshot();
});
