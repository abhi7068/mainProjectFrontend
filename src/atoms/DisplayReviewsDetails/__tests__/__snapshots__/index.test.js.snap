// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
      style="color: black;"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <div
          aria-label="recipe"
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-5 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          Madan Murari
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        />
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <span
        aria-label="5 Stars"
        class="MuiRating-root MuiRating-readOnly"
        role="img"
        style="color: rgb(228, 0, 70);"
      >
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        style="font-size: 1.5em; text-align: left;"
      >
        Nice Picture
      </p>
    </div>
  </div>
</DocumentFragment>
`;
